runtime/Cpp/runtime/src/misc/IntervalSet.cpp

```diff
@@ -13,52 +13,31 @@
 using namespace antlr4;
 using namespace antlr4::misc;
 
-IntervalSet const IntervalSet::COMPLETE_CHAR_SET = []() {
-  IntervalSet complete = IntervalSet::of(Lexer::MIN_CHAR_VALUE, Lexer::MAX_CHAR_VALUE);
-  complete.setReadOnly(true);
-  return complete;
-}();
+IntervalSet const IntervalSet::COMPLETE_CHAR_SET = 
+    IntervalSet::of(Lexer::MIN_CHAR_VALUE, Lexer::MAX_CHAR_VALUE);
 
-IntervalSet const IntervalSet::EMPTY_SET = []() {
-  IntervalSet empty;
-  empty.setReadOnly(true);
-  return empty;
-}();
+IntervalSet const IntervalSet::EMPTY_SET;
 
-IntervalSet::IntervalSet() {
-  InitializeInstanceFields();
+IntervalSet::IntervalSet() : _intervals() {
 }
 
-IntervalSet::IntervalSet(const std::vector<Interval> &intervals) : IntervalSet() {
-  _intervals = intervals;
+IntervalSet::IntervalSet(const IntervalSet &set) : _intervals(set._intervals) {
 }
 
-IntervalSet::IntervalSet(const IntervalSet &set) : IntervalSet() {
-  addAll(set);
+IntervalSet::IntervalSet(IntervalSet&& set) : _intervals(std::move(set._intervals)) {
 }
 
-IntervalSet::IntervalSet(int n, ...) : IntervalSet() {
-  va_list vlist;
-  va_start(vlist, n);
-
-  for (int i = 0; i < n; i++) {
-    add(va_arg(vlist, int));
-  }
+IntervalSet::IntervalSet(std::vector<Interval>&& intervals) : _intervals(std::move(intervals)) {
```

Comment:
The varadic code starts on line 42 of IntervalSet.h, but I think you found it from your later comment.

The constructor that takes a vector<Interval> rvalue reference is a private constructor used internally. If we want to use a constant here it will be local to this class and can be an explicit decision.

Removing type-unsafe va_args code certainly seems like an improvement to me: The old code is likely to fail outright, especially on machines where sizeof(int) != sizeof(ssize_t). The new code is type-safe, provides compile-time diagnostics, and allows single and multi-value intervals. For the same arguments I don't see a semantic change (excluding failures because of va_args usage problems).